File: python/functionTest.py

def test():
	def f():
		return 1
	print(f())

def test2():
	def f(a):
		return a + 1
	print(f(2))

def test3():
	def f(a, b):
		return a + b
	print(f(2, 3))

def test4():
	def f(*a):
		s=0
		for x in a:
			s += x
		return s
	print(f(2, 3, 4, 5))

def test5():
	def f(**a):
		s=0
		for v in a.values():
			s += v
		return s
	print(f(a=2, b=4, x=6, y=8))

def test6():
	def f(a,*,b):
		return a + b
	print(f(a=2, b=4))

def test7():
	f = lambda a, b : a + b
	print(f(1,2))

test7()

File: swift/functionTest.swift

func test(){
	func f(s: String) ->String{
	    return "apple " + s
	}
	let a=f(s: "banana")
	print(a)
}

func test2(){
    func f(fruit s: String) ->String{
	    return "apple " + s
	}
	let a=f(fruit: "banana")
	print(a)
}

func test3(){
    func f(_ s: String) ->String{
	    return "apple " + s
	}
	let a=f("banana")
	print(a)
}

func test4(){
    func f(_ s: inout String){
	    return s = "apple " + s
	}
	var a = "banana"
	f(&a)
	print(a)
}

func test5(){
    func f(_ array: String...){
	    for k in array {
	        print(k)
	    }
	}
	f("apple", "banana", "orange")
}

func test6(){
    func f(_ array: [String]){
	    for k in array {
	        print(k)
	    }
	}
	f(["apple", "banana", "orange"])
}

